editor/src/clj/editor/compute_program.clj

```diff
@@ -0,0 +1,177 @@
+  (mapv (fn [sampler]
+          (assoc sampler :name-hash (murmur/hash64 (:name sampler))))
+        samplers))
+
+(defn- prop-resource-error [_node-id prop-kw prop-value prop-name resource-ext]
+  (validation/prop-error :fatal _node-id prop-kw validation/prop-resource-ext? prop-value resource-ext prop-name))
+
+(g/defnk produce-build-targets [_node-id base-pb-msg resource shader-source-info program]
+  (or (g/flatten-errors
+        (prop-resource-error _node-id :program program "Compute Program" "cp"))
+      (let [compile-spirv true
+            compute-shader-build-target (code.shader/make-shader-build-target shader-source-info compile-spirv 0)
+            dep-build-targets [compute-shader-build-target]
+            compute-program-desc-with-build-resources (assoc base-pb-msg
+                                                        :program (:resource compute-shader-build-target)
+                                                        :samplers (build-target-samplers (:samplers base-pb-msg)))]
+        [(bt/with-content-hash
+           {:node-id _node-id
+            :resource (workspace/make-build-resource resource)
+            :build-fn build-compute-program
+            :user-data {:compute-program-desc-with-build-resources compute-program-desc-with-build-resources}
+            :deps dep-build-targets})])))
+
+(g/defnode ComputeProgramNode
+  (inherits resource-node/ResourceNode)
+
+  (property name g/Str (dynamic visible (g/constantly false)))
+
+  (property program resource/Resource
+            (dynamic visible (g/constantly false))
+            (value (gu/passthrough program-resource))
+            (set (fn [evaluation-context self old-value new-value]
+                   (project/resource-setter evaluation-context self old-value new-value
+                                            [:resource :program-resource]
+                                            [:shader-source-info :shader-source-info]))))
+  (property constants g/Any (dynamic visible (g/constantly false)))
+  (property samplers g/Any
+            (dynamic visible (g/constantly false))
+            (value (gu/passthrough samplers)))
+
+  (input program-resource resource/Resource)
+  (input shader-source-info g/Any)
+
+  (output form-data g/Any :cached produce-form-data)
+  (output base-pb-msg g/Any produce-base-pb-msg)
+  (output save-value g/Any (gu/passthrough base-pb-msg))
+  (output build-targets g/Any :cached produce-build-targets)
+  (output samplers [g/KeywordMap] (gu/passthrough samplers)))
+
+(defn load-compute-program [project self resource pb]
+  (concat
+    (g/set-property self :program (workspace/resolve-resource resource (:program pb)))
+    (g/set-property self :constants (material/hack-downgrade-constants (:constants pb)))
+    (g/set-property self :samplers (:samplers pb))))
+
+(defn register-resource-types [workspace]
+  (resource-node/register-ddf-resource-type workspace
+    :ext "compute_program"
+    :label "Compute"
```

Comment:
Should we maybe use `.compute` for the file extension?

Reply:
Ah alright, yes I'm down with naming it .compute!